### ISSUE TITLE
artifactName is not work as expected.

### ISSUE THREAD

**Issue by miaulightouch** [electron-userland/electron-builder#2258]
* **Version**: 19.43.3

* **Target**: linux/win

here's my config:
```javascript
    "win": {
      "artifactName": "${productName}-${version}-${os}-${arch}.${ext}",
      "target": [{
        "target": "zip",
        "arch": ["x64", "ia32"]
      }]
    },
    "linux": {
      "artifactName": "${productName}-${version}-${os}-${arch}.${ext}",
      "target": [{
        "target": "AppImage",
        "arch": ["x64", "ia32"]
      }]
    }
```
but CI (AppVeyor/Travis) generated filename is 
```
MasterVyrn-0.2.0-linux-i386.AppImage
MasterVyrn-0.2.0-linux.AppImage        (error: no arch)
mastervyrn-0.2.0-ia32-win.zip          (error: pattern order wrong)
mastervyrn-0.2.0-win.zip               (error: no arch)
```

**Comment by develar** [comment]
BTW, for Windows consider to use portable.

**Comment by develar** [comment]
You can also specify `artifactName` in the `build`, no need to repeat in the `win`/`linux`

**Comment by miaulightouch** [comment]
yes, I know, I just must ensure `artifactName`  work properly

**Comment by develar** [comment]
Fixed in 19.44.0
